R/secondary_weights.R

```diff
@@ -208,6 +211,16 @@ secondary_weights <- function(secondary_raster, grid = era5_grid, extent = "full
 
   }
 
+  # Reproject if necessary
+  if(terra::crs(secondary_raster, TRUE) != terra::crs(grid, TRUE)){
+
+    message(crayon::red("Warning: reprojecting secondary_raster to match grid, which may affect data"))
+    secondary_raster <- terra::project(secondary_raster, clim_raster)
+    terra::crs(secondary_raster) <- terra::crs(clim_raster, TRUE)
```

Comment:
i think that `terra::project()` should automatically change the label (this seemed to happen when i tested the code), but i don't think it would lead to any errors? maybe unless the label was incorrect from the beginning?